core/scripts/liquidity-mining/CalculateBalancerLPProviders.js

```diff
@@ -0,0 +1,155 @@
+// This script calculates $UMA liquidity mining rewards for Balancer Pools. This is done with the following process:
+// -> Define the starting and ending blockheight of each week. Both are chosen as the block with the closest timestamp
+// to a fixed weekly time(e.g.2020-06-06 00:00pm UTC).
+// -> Define snapshot blocks (eg every 64 blocks ~15min). For each snapshot block, calculate the proportional liquidity
+// provided by for each liquidity provider to the single whitelisted pool.
+// -> For each snapshot block, calculate the $UMA rewards to be received by each liquidity provider based on the target weekly distribution.
+
+// Example usage from core: node ./scripts/liquidity-mining/CalculateBalancerLPProviders.js --poolAddress="0x0099447ef539718bba3c4d4d4b4491d307eedc53" --fromDate="2020-07-06" --toDate="2020-07-13" --week=1
```

Comment:
I am happy this is not a truffle exec script

Reply:
ye, I wanted to give that a go. Ideally, we can use a similar method to refactor the liquidator bots to simply be a node script.